.github/scripts/create_pr_for_staged_changes.sh

```diff
@@ -0,0 +1,114 @@
+
+# Sets GitHub Action with error message to ease troubleshooting
+function raise_validation_error() {
+    echo "::error file=${FILENAME}::$1"
+    exit 1
+}
+
+function debug() {
+    echo "::debug::$1"
+}
+
+function notice() {
+    echo "::notice file=${FILENAME}::$1"
+}
+
+function has_required_config() {
+    # Default GitHub Actions Env Vars: https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+    debug "Do we have required environment variables?"
+    test -z "${TEMP_BRANCH_PREFIX}" && raise_validation_error "TEMP_BRANCH_PREFIX env must be set to create a PR"
+    test -z "${GH_TOKEN}" && raise_validation_error "GH_TOKEN env must be set for GitHub CLI"
+    test -z "${GITHUB_RUN_ID}" && raise_validation_error "GITHUB_RUN_ID env must be set to trace Workflow Run ID back to PR"
+    test -z "${GITHUB_SERVER_URL}" && raise_validation_error "GITHUB_SERVER_URL env must be set to trace Workflow Run ID back to PR"
+    test -z "${GITHUB_REPOSITORY}" && raise_validation_error "GITHUB_REPOSITORY env must be set to trace Workflow Run ID back to PR"
+
+    set_environment_variables
+}
+
+function set_environment_variables() {
+    WORKFLOW_URL="${GITHUB_SERVER_URL}"/"${GITHUB_REPOSITORY}"/actions/runs/"${GITHUB_RUN_ID}" # e.g., heitorlessa/aws-lambda-powertools-test/actions/runs/4913570678
+    TEMP_BRANCH="${TEMP_BRANCH_PREFIX}"-"${GITHUB_RUN_ID}"                                     # e.g., ci-changelog-4894658712
+
+    export readonly WORKFLOW_URL
+    export readonly TEMP_BRANCH
+}
+
+function has_anything_changed() {
+    debug "Is there an update to the source code?"
+    HAS_ANY_SOURCE_CODE_CHANGED="$(git status --porcelain)"
+
+    test -z "${HAS_ANY_SOURCE_CODE_CHANGED}" && echo "Nothing to update" && exit 0
+}
+
+function create_temporary_branch_with_changes() {
+    debug "Creating branch ${TEMP_BRANCH}"
+    git checkout -b "${TEMP_BRANCH}"
+
+    debug "Committing staged files: $*"
+    git add "$@"
+    git commit -m "${COMMIT_MSG}"
+
+    debug "Creating branch remotely"
+    git push origin "${TEMP_BRANCH}"
+}
+
+function create_pr() {
+    debug "Creating PR against ${BRANCH} branch"
+    NEW_PR_URL=$(gh pr create --title "${PR_TITLE}" --body "${PR_BODY}: ${WORKFLOW_URL}" --base "${BRANCH}") # e.g, https://github.com/awslabs/aws-lambda-powertools/pull/13
+
+    # greedy remove any string until the last URL path, including the last '/'. https://opensource.com/article/17/6/bash-parameter-expansion
+    NEW_PR_ID="${NEW_PR_URL##*/}" # 13
+    export NEW_PR_URL
+    export NEW_PR_ID
+}
+
+function close_duplicate_prs() {
+    debug "Do we have any duplicate PRs?"
+    DUPLICATE_PRS=$(gh pr list --search "${PR_TITLE}" --json number --jq ".[] | select(.number != ${NEW_PR_ID}) | .number") # e.g, 13\n14
+
+    debug "Closing duplicated PRs if any"
+    echo "${DUPLICATE_PRS}" | xargs -L1 gh pr close --delete-branch --comment "Superseded by #${NEW_PR_ID}"
+    export readonly DUPLICATE_PRS
+}
```

Comment:
Super cool this check to avoid orphans branches in the repository and getting messy. :D